File: .sdk-repos/orchestration-cluster-api-csharp/examples/Decision.cs
// Compilable usage examples for decision operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class DecisionExamples
{
    #region EvaluateDecisionById
    // <EvaluateDecisionById>
    public static async Task EvaluateDecisionByIdExample(DecisionDefinitionId decisionDefinitionId)
    {
        using var client = CamundaClient.Create();

        var result = await client.EvaluateDecisionAsync(new DecisionEvaluationById
        {
            DecisionDefinitionId = decisionDefinitionId,
        });

        Console.WriteLine($"Decision output: {result.Output}");
    }
    // </EvaluateDecisionById>
    #endregion EvaluateDecisionById

    #region EvaluateDecisionByKey

    // <EvaluateDecisionByKey>
    public static async Task EvaluateDecisionByKeyExample(DecisionDefinitionKey decisionDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.EvaluateDecisionAsync(new DecisionEvaluationByKey
        {
            DecisionDefinitionKey = decisionDefinitionKey,
        });

        Console.WriteLine($"Decision output: {result.Output}");
    }
    // </EvaluateDecisionByKey>
    #endregion EvaluateDecisionByKey

    #region GetDecisionDefinition

    // <GetDecisionDefinition>
    public static async Task GetDecisionDefinitionExample(DecisionDefinitionKey decisionDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetDecisionDefinitionAsync(
            decisionDefinitionKey);

        Console.WriteLine($"Decision definition: {result.Name}");
    }
    // </GetDecisionDefinition>
    #endregion GetDecisionDefinition

    #region GetDecisionDefinitionXml

    // <GetDecisionDefinitionXml>
    public static async Task GetDecisionDefinitionXmlExample(DecisionDefinitionKey decisionDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetDecisionDefinitionXmlAsync(
            decisionDefinitionKey);

        Console.WriteLine($"XML: {result}");
    }
    // </GetDecisionDefinitionXml>
    #endregion GetDecisionDefinitionXml

    #region SearchDecisionDefinitions

    // <SearchDecisionDefinitions>
    public static async Task SearchDecisionDefinitionsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchDecisionDefinitionsAsync(
            new DecisionDefinitionSearchQuery());

        foreach (var dd in result.Items)
        {
            Console.WriteLine($"Decision definition: {dd.Name}");
        }
    }
    // </SearchDecisionDefinitions>
    #endregion SearchDecisionDefinitions

    #region GetDecisionInstance

    // <GetDecisionInstance>
    public static async Task GetDecisionInstanceExample(DecisionEvaluationInstanceKey decisionEvaluationInstanceKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetDecisionInstanceAsync(
            decisionEvaluationInstanceKey);

        Console.WriteLine($"Decision instance: {result.DecisionDefinitionId}");
    }
    // </GetDecisionInstance>
    #endregion GetDecisionInstance

    #region SearchDecisionInstances

    // <SearchDecisionInstances>
    public static async Task SearchDecisionInstancesExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchDecisionInstancesAsync(
            new DecisionInstanceSearchQuery());

        foreach (var di in result.Items)
        {
            Console.WriteLine($"Decision instance: {di.DecisionDefinitionId}");
        }
    }
    // </SearchDecisionInstances>
    #endregion SearchDecisionInstances

    #region DeleteDecisionInstance

    // <DeleteDecisionInstance>
    public static async Task DeleteDecisionInstanceExample(DecisionEvaluationKey decisionEvaluationKey)
    {
        using var client = CamundaClient.Create();

        await client.DeleteDecisionInstanceAsync(
            decisionEvaluationKey,
            new DeleteDecisionInstanceRequest());
    }
    // </DeleteDecisionInstance>
    #endregion DeleteDecisionInstance

    #region GetDecisionRequirements

    // <GetDecisionRequirements>
    public static async Task GetDecisionRequirementsExample(DecisionRequirementsKey decisionRequirementsKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetDecisionRequirementsAsync(
            decisionRequirementsKey);

        Console.WriteLine($"DRD: {result.DecisionRequirementsName}");
    }
    // </GetDecisionRequirements>
    #endregion GetDecisionRequirements

    #region GetDecisionRequirementsXml

    // <GetDecisionRequirementsXml>
    public static async Task GetDecisionRequirementsXmlExample(DecisionRequirementsKey decisionRequirementsKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetDecisionRequirementsXmlAsync(
            decisionRequirementsKey);

        Console.WriteLine($"XML: {result}");
    }
    // </GetDecisionRequirementsXml>
    #endregion GetDecisionRequirementsXml

    #region SearchDecisionRequirements

    // <SearchDecisionRequirements>
    public static async Task SearchDecisionRequirementsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchDecisionRequirementsAsync(
            new DecisionRequirementsSearchQuery());

        foreach (var drd in result.Items)
        {
            Console.WriteLine($"DRD: {drd.DecisionRequirementsName}");
        }
    }
    // </SearchDecisionRequirements>
    #endregion SearchDecisionRequirements
}

File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/decision.py
# Compilable usage examples for decision evaluation and search.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    DecisionDefinitionId,
    DecisionDefinitionKey,
    DecisionDefinitionSearchQuery,
    DecisionEvaluationByID,
    DecisionEvaluationByKey,
    Unset,
)


# region EvaluateDecisionByKey
def evaluate_decision_by_key_example(decision_definition_key: DecisionDefinitionKey) -> None:
    client = CamundaClient()

    result = client.evaluate_decision(
        data=DecisionEvaluationByKey(
            decision_definition_key=decision_definition_key,
        )
    )

    print(f"Decision key: {result.decision_definition_key}")
# endregion EvaluateDecisionByKey


# region EvaluateDecisionById
def evaluate_decision_by_id_example(decision_definition_id: DecisionDefinitionId) -> None:
    client = CamundaClient()

    result = client.evaluate_decision(
        data=DecisionEvaluationByID(
            decision_definition_id=decision_definition_id,
        )
    )

    print(f"Decision key: {result.decision_definition_key}")
# endregion EvaluateDecisionById


# region SearchDecisionDefinitions
def search_decision_definitions_example() -> None:
    client = CamundaClient()

    result = client.search_decision_definitions(
        data=DecisionDefinitionSearchQuery()
    )

    if not isinstance(result.items, Unset):
        for definition in result.items:
            print(f"Decision: {definition.decision_definition_id}")
# endregion SearchDecisionDefinitions


# region GetDecisionDefinition
def get_decision_definition_example(decision_definition_key: DecisionDefinitionKey) -> None:
    client = CamundaClient()

    definition = client.get_decision_definition(
        decision_definition_key=decision_definition_key,
    )

    print(f"Decision: {definition.decision_definition_id}")
# endregion GetDecisionDefinition
